polling_stations/apps/data_importers/data_types.py

```diff
@@ -69,22 +69,44 @@ def council_id(self):  # TODO Deal with old_to_new council_ids map
         for e in self.elements:
             return e.council.council_id
 
+    def save(self):
+        districts_db = []
+        for district in self.elements:
+            record = PollingDistrict(
+                name=district.name,
+                council=district.council,
+                internal_council_id=district.internal_council_id,
+                extra_id=district.extra_id,
+                area=district.area,
+                polling_station_id=district.polling_station_id,
+            )
+            districts_db.append(record)
+        PollingDistrict.objects.bulk_create(districts_db)
+
     def get_polling_station_lookup(self):
         # for each address, build a lookup of polling_station_id -> set of uprns
         polling_station_lookup = {}
-        uprn_hash_table = get_uprn_hash_table(self.council_id)
-        for district in self.elements:
-            district_geom = GEOSGeometry(district.area)
-            polling_station_id = district.polling_station_id
-
+        cursor = connection.cursor()
+        cursor.execute(
+            """ 
+                SELECT a.uprn, d.polling_station_id
+                FROM addressbase_address a
+                    JOIN addressbase_uprntocouncil u
+                    ON a.uprn = u.uprn
+                    JOIN pollingstations_pollingdistrict d
+                    ON ST_Covers(d.area, a.location)
+                WHERE d.council_id=%s
+                AND u.lad=%s
+            """,
+            [self.council_id, self.council_id],
+        )
```

Comment:
I'm sure this can be done with the orm, but this was easier for me to write for now grin .